docs/reference/api/widgets/imageview.rst

```diff
@@ -17,7 +17,14 @@ Usage
 
     import toga
 
-    view = toga.ImageView(id='view1', image=my_image)
+    my_image = toga.Image(self.paths.app / "brutus.png")
+    view = toga.ImageView(my_image)
+
+Notes
+-----
+
+* The default size of the view is the size of the image, or 0x0 if ``image`` is
+  ``None``.
```

Comment:
There are a few existing issues about ImageView sizing, e.g.:
* https://github.com/beeware/toga/issues/1795
* https://github.com/beeware/toga/issues/1923

Currently the only reliable workaround is to set both `width` and `height` in the style. When you do that in CSS, the image is resized to those exact dimensions, disregarding its aspect ratio. So we should probably do the same, and if we need to make this configurable in future, we can add support for the [`object-fit`](https://developer.mozilla.org/en-US/docs/Web/CSS/object-fit) attribute.

However, a more common use case is probably to set the width *or* height, and let the other one be determined by the image's aspect ratio. CSS supports this, but I guess Pack would just set the given dimension, and leave the other one at its instrinsic size.

This is not ideal, but it's at least easy for the programmer to understand. So my suggestion is:
* Add read-only `width` and `height` properties to the `Image` class, to enable the programmer to do this kind of scaling manually.
* Create an issue to remind us to implement the CSS behavior in the future.

I've only mentioned `width` and `height` here, but we should also consider what would happen with `flex`.

Reply:
It turns out supporting aspect-ratio preservation in the width-only and height-only case actually isn't that hard (at least, on Cocoa and iOS... I'm sure GTK will make the regret those words shortly). We have access to both the image size and the style in `rehint()`; so we can set the intrinsic size to something that is derived from those properties. 

We don't *strictly* need to expose public width/height attributes on Image - the ImageView can access the native image object associated with the native image view - but it also doesn't hurt to expose them either. Regardless, they're generally useful attributes with trivial implementations, so we might as well expose them.

As for flex: I think the most compliant thing we can do is:
* set the intrinsic size of the the image to be `at_least(0)` in the flow axis (so - height if it's a column box, width on a row box).
* set the intrinsic size of the image in the cross axis to be `at_least()` the explicitly provided size, or the image native size if no explicit size is provided.
* Modify the image rendering to preserve aspect ratio unless *both* height and width are specified.

AFAICT, this matches HTML in all cases *except* the case where only flex=1 is defined. In that case, I *think* HTML tries to use the size of the container to present an aspect-ratio preserving size - but we don't have access to the size of the container, so we'll have to live with (and presumably document) the discrepancy.